File: api/src/main/java/api/dtos/requests/UserAuthenticationRequestDTO.java
package api.dtos.requests;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;

public record UserAuthenticationRequestDTO(
        @NotBlank(message = "Informe o seu email.")
        @Size(min = 5, max = 80, message = "O email informado é inválido.")
        @Email(message = "O email informado é inválido.")
        String email,
        @NotBlank(message = "Informe a sua senha.")
        @Size(min = 6, message = "A senha informada é inválida.")
        String password
) {
    public Authentication convert() {
        return new UsernamePasswordAuthenticationToken(
                email(),
                password()
        );
    }
}

File: web/src/core/dtos/requests/user-authentication-request.dto.ts
import { z } from "zod";

interface UserAuthenticationRequestDTO {
  email: string;
  password: string;
}

const validationSchema = z.object({
  email: z.string()
    .email({ message: "O email informado é inválido." })
    .min(5, { message: "O email deve conter no mínimo 5 caracteres." })
    .max(80, { message: "O email deve conter no máximo 80 caracteres." }),
  password: z.string()
    .min(6, { message: "A senha deve conter no mínimo 6 caracteres." })
});

export default UserAuthenticationRequestDTO;
export { validationSchema };
